app/master/cluster_master.py

```diff
@@ -151,8 +161,7 @@ def _disconnect_slave(self, slave):
         """
         # Mark slave dead. We do not remove it from the list of all slaves. We also do not remove it from idle_slaves;
         # that will happen during slave allocation.
-        slave.set_is_alive(False)
-        slave.current_build_id = None
+        slave.disconnect()
```

Comment:
Ah, yes, that confused me. So the _disconnect_slave() method in this class simply updates the master's bookeaping on that particular slave.